test/harry/main/org/apache/cassandra/harry/model/ASTSingleTableModel.java

```diff
@@ -765,7 +767,10 @@ private List<PrimaryKey> filter(LookupContext ctx, List<BytesPartitionState> par
         if (partitions.isEmpty()) return Collections.emptyList();
         List<PrimaryKey> matches = new ArrayList<>();
         for (BytesPartitionState p : partitions)
+        {
+            if (!ctx.include(p)) continue;
```

Comment:
`ASTSingleTableModelTest` passes with this removed

Reply:
this leads to calling https://github.com/apache/cassandra/pull/3864/files#r1940167860.  This is just being extra defensive as the write path is supposed to delete the partitions once `shouldDelete() == true`.  This is only here if there exists any other write path bugs